File: finpilot-ai-frontend/src/types/notification.ts
export type NotificationType =
  | "LOAN_REQUEST"
  | "LOAN_APPROVED"
  | "LOAN_REJECTED"
  | "LOAN_PAYMENT"
  | "LOAN_CLOSED"
  | "LARGE_TRANSACTION"
  | "UNUSUAL_TRANSACTION"
  | "LOW_BALANCE"
  | "FINANCIAL_INSIGHT"
  | "AI_RECOMMENDATION"
  | "SYSTEM";

export interface Notification {
  id: string;
  user_id: string;
  notification_type: NotificationType | string;
  title: string;
  message: string;
  is_read: boolean;
  created_at: string;
}

export interface NotificationListResponse {
  notifications: Notification[];
  unread_count: number;
}

export interface Notification {
  id: string;
  user_id: string;
  notification_type: string;
  title: string;
  message: string;
  is_read: boolean;
  created_at: string;
}

export interface NotificationListResponse {
  notifications: Notification[];
  unread_count: number;
}

export interface MarkAllNotificationsResponse {
  message: string;
  updated_count: number;
}

File: app/schemas/notification.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, ConfigDict


class NotificationCreate(BaseModel):
    customer_id: UUID
    notification_type: str
    title: str
    message: str


class NotificationResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    user_id: UUID
    notification_type: str
    title: str
    message: str
    is_read: bool
    created_at: datetime


class NotificationListResponse(BaseModel):
    notifications: list[NotificationResponse]
    unread_count: int